client/spa-pages/components/InstructionsPage.tsx

```diff
@@ -1,32 +1,83 @@
-import { Flex, Text, Container, VStack } from "@chakra-ui/react";
+import { Flex, Text, Container, VStack, filter } from "@chakra-ui/react";
+import { useSheetyData } from "hooks/useRecyclableItemList";
+import { TInstructions } from "api/sheety/types";
 import { BasePage } from "layouts/BasePage";
 import { AccordionComp, InstructionsHeader } from "components/instructions";
-import { Dispatch, SetStateAction } from "react";
+import { Dispatch, SetStateAction, useRef, useEffect, useState } from "react";
 import { Pages } from "spa-pages/pageEnums";
+import { useUserInputs } from "hooks/useUserSelection";
+import { StickyFooterInstructions } from "components/footer/StickyFooterInstructions";
+import { useWindowDimensions } from "hooks/useWindowDimensions";
+import { NAVBAR_HEIGHT } from "components/nav/NavHeader";
 
 type Props = {
 	setPage: Dispatch<SetStateAction<Pages>>;
 };
 
-const items = [
-	{ title: "Title", method: "Method", imgURL: "imgUrl", contents: ["1", "2"] },
-	{ title: "Title 2", method: "Method 2", imgURL: "imgUrl 2", contents: ["3", "4"] },
-];
+type itemType = {
+	name: string;
+	method: string;
+};
+
+const itemFilter = (items: itemType[], instructions: TInstructions[]) => {
+	const filteredSelections = items.filter((item) => item.method !== "");
+	const filteredInstructions = instructions
+		.filter((instructionItem) =>
+			filteredSelections.some((selectedItem) => selectedItem.name === instructionItem.name),
+		)
+		.map((instructionItem) => {
+			const contents = [];
+			contents.push(instructionItem.step1);
+			if (instructionItem.step2 !== undefined) {
+				contents.push(instructionItem.step2);
+			}
+			if (instructionItem.step3 !== undefined) {
+				contents.push(instructionItem.step3);
+			}
+			return {
+				title: instructionItem.name,
+				method: "",
+				contents,
+			};
+		})
+		.map((instrItem) => {

```

Comment:
can this map be merged with the previous map?

Reply:
Ah yes, managed to do so